src/tracing/esp32_trace/include/matter/tracing/macros_impl.cpp

```diff
@@ -28,17 +38,47 @@ namespace Insights {
                        heap_caps_get_free_size(MALLOC_CAP_8BIT));                                                                  \
     } while (0)
 
+void ESP32Backend::InitializeBlacklistHashes()
+{
+    for (int i = 0; i < blacklistSize; i++)
+    {
+        if (blackListStrings[i] == NULL)
+            break;
+        size_t data_length = strlen(blackListStrings[i]);
+        uint8_t sha1_hash[20];
+        CHIP_ERROR error        = chip::Crypto::Hash_SHA1((const uint8_t *) blackListStrings[i], data_length, sha1_hash);
+        uint64_t truncated_hash = 0;
+        for (int j = 0; j < 8; j++)
+        {
+            truncated_hash <<= 8;
+            truncated_hash |= sha1_hash[j];
+        }
+        blacklistHashes[i] = truncated_hash;
+    }
+}
+
 ESP32Backend::ESP32Backend(const char * str, ...)
 {
     va_list args;
     va_start(args, str);
     mlabel = str;
     mgroup = va_arg(args, const char *);
-    LOG_HEAP_INFO(mlabel, mgroup, "Entry");
+    if (!isBlacklistInitialized)
+    {
+        InitializeBlacklistHashes();
+        isBlacklistInitialized = true;
+    }
+    if (!isBlacklistedString(mgroup))
+    {
+        LOG_HEAP_INFO(mlabel, mgroup, "Entry");
```

Comment:
You seem to have the ability to implement begin/end macros.

You could also implement a `chip::tracing::Backend` instead of overriding all macros and such an implementation would be type-safe (it provides label and group as arguments and no VA_ARGS).